### ISSUE TITLE
JAWS announces "blank" when pressing caret specific keys on div inside role=application

### ISSUE THREAD

**Issue by asuceava** [FreedomScientific/VFO-standards-support#487]
### Summary

I am working on a custom widget where items can be navigated and (among other actions) deleted. Because it is very complex, we had to use role="application" for it. The delete action is intuitively mapped to the Delete key. Whenever the key is pressed, the delete action itself works, but JAWS annoyingly announces "blank" every time, which is a bit confusing since there is no actual editable content, just a plain focused div. After connecting the dots, I also attempted to use Backspace and Ctrl + Left/Right arrow, and unsurprisingly they produce the same result.

I minimized the use case in a basic code snippet. Get it here: [test-del-bksp-ctrl_lr.zip](https://github.com/FreedomScientific/VFO-standards-support/files/5854866/test-del-bksp-ctrl_lr.zip)

 Steps to reproduce:
 1. Open attached snippet
 2. Press Tab once to focus on the div
 3. While focused, try pressing Delete, Backspace, or Ctrl + Left/Right arrow

### Expected result

Because the focused element is not a text input element, Delete, Backspace and Ctrl + Left/Right are captured and handled by the browser and do not announce any text/caret-specific output.

### Actual result

Delete, Backspace and Ctrl + Left/Right are captured and handled by the browser, but are also captured by JAWS and announce "blank", even though there is no editable content.

### Example
See attached snippet.

### Additional Information
Also attempted to use other non-input elements (e.g span), but the result is the same.
Attempted to intercept the event as early as possible (on Window in the event capture phase), but it's clearly intercepted by JAWS before that.

#### JAWS version and build number
2019 (1912.1 ILM)
2021 (2011.16 ILM)
 
#### Operating System and version
OS: Windows 10 Enterprise 
Version: 1809
Build: 17763.1518

#### Browser and version:
Chrome (tested versions: 75 through 88)
Edge Chromium (87.0.664.75)

**Comment by JAWS-test** [comment]
The element that receives the focus needs a role (e.g. an ARIA role via role=...) and a label. If I add both in your example, the problem does not occur anymore. However, the new problem is that the first letter of the label is output as an accesskey after pressing Delete or Backspace. This is a JAWS bug.

**Comment by asuceava** [comment]
@JAWS-test, Thanks for the quick reply.

I understand the need for a role + label when an actual item's element is focused (well, mostly the label, because the focused element can also represent something non-generic, or a business entity), but there is another widely used practice in complex widgets where the focus is kept on an element and ALL the keyboard handling is done there. This latter case is usually implemented in tandem with a live region that outputs customized information. I deduced after a closer inspection that this is also how the Google Docs suite currently works, and we also have a few widgets in our application that use this approach.

I noticed that JAWS identifies certain widget design patterns inside a role="application" container as well, which is brilliant, but my belief is that they should not be required by all focused elements.

Based on your reply, I attempted something a little less orthodox in my snippet:
<div role="presentation" aria-label="&#8203;"></div>

... where &#8203; is the unicode for a zero-width space. Apparently JAWS doesn't know how to read it, so it's now silent.

**Comment by JAWS-test** [comment]
> but there is another widely used practice in complex widgets where the focus is kept on an element and ALL the keyboard handling is done there. This latter case is usually implemented in tandem with a live region that outputs customized information. I deduced after a closer inspection that this is also how the Google Docs suite currently works, and we also have a few widgets in our application that use this approach

1. because Google does something this way does not mean it is right
2. live regions are very [error-prone](https://github.com/FreedomScientific/VFO-standards-support/issues?q=is%3Aissue+live%2Bregion) and should not be used for this purpose 
3. WCAG 4.1.2 requires name, role and value (independent of live regions).

**Comment by asuceava** [comment]
Thanks for the link at point 3. It gave me some more insight into this particular topic.

As for 1 and 2, I agree, but not fully:
- Having a correct ARIA markup is indeed desired, and will ensure a long term compatibility with most assistive technologies as they evolve.
- When making our (extremely complex) app accessible we did start with the ARIA (1.0 at the time) best practices, but along the way we encountered various problems with the suggested widget design patterns not being properly read out, ranging from simpler ones to more complex scenarios like spreadsheet-like editable tables and multi-level tab boxes. For these we had to use alternative methods, such as the role="application" + live regions to simulate the expected screen reader output.
- To achieve a cross-browser-cross-screen-reader solution is currently impossible using just ARIA once a certain complexity threshold is reached.
- I personally see what Google did purely as a market positioning move, aka to have a working product while other developers are still waiting for the implementation of the standards by all the actors.
- A high number of bugs with live regions can also mean a very frequent use. Of course as user agents and assistive technologies continue to evolve, their need will also diminish.

**Comment by JAWS-test** [comment]
I agree that not everything can be done with ARIA or HTML. But a lot can be done with ARIA and HTML. And what can't be done with ARIA + HTML could be redesigned to work with ARIA + HTML. E.g. you can use aria-roledescription for custom roles, but use a role in the background that is similar and causes form mode, so no role=application is needed. Instead of live regions, aria-label or aria-describedby, aria-errormessage or aria-keyshortcuts can be used to convey specific information. 

I can't say more, because I don't know your application. By the way, the output of "blank" is a very minor problem. JAWS often outputs "blank" and this should not be a major issue.
